web/onos-gui/src/app/onos-config/config-view/config-view.component.ts

```diff
@@ -39,23 +46,25 @@ export class ConfigViewComponent implements OnInit, OnChanges {
     type: string;
     updated: number = 0;
     changeIds: string[] = [];
-    changeIdsVisible: Map<string, boolean>;
-    hasStateData: boolean = true;
-    hasOpData: boolean;
+    changeIdsVisible = new Map<string, boolean>();
+    hasOpStateData: boolean = true;
+    create_pending: string = '';
+    create_pending_confirm: string = '';
+
+    // Constants - have to declare a viable to hold a constant so it can be used in HTML(?!?!)
+    public OPSTATE = OPSTATE;
 
     constructor(
         private diags: OnosConfigDiagsService,
+        private pending: PendingNetChangeService,
         protected ar: ActivatedRoute,
         protected is: IconService,
     ) {
         this.is.loadIconDef('checkMark');
         this.is.loadIconDef('xMark');
-        this.changeIdsVisible = new Map();
-        if (this.hasStateData) {
-            this.changeIdsVisible['state'] = false;
-        }
-        if (this.hasOpData) {
-            this.changeIdsVisible['operational'] = false;
+        this.is.loadIconDef('plus');
```

Comment:
do we have an icon definition somewhere, is it worth to extract this and define the icon name as constant and just re-use it here ?

Reply:
the IconService and the definitions are all in ONOS classic
https://github.com/opennetworkinglab/onos/blob/master/web/gui2-fw-lib/projects/gui2-fw-lib/src/lib/svg/icon.service.ts
This will have to be updated at some stage when we want new icons - for the moment I'm running off the existing ones